modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java

```diff
@@ -413,6 +456,24 @@ public Account patchAccount(Long accountId, Account account)
 		return _toAccount(accountEntry);
 	}
 
+	@Override
+	public Account patchAccountByExternalReferenceCode(
+			String externalReferenceCode, Account account)
+		throws Exception {
+
+		AccountEntry accountEntry =
+			_accountEntryService.fetchAccountEntryByExternalReferenceCode(
+				contextCompany.getCompanyId(), externalReferenceCode);
+
+		if (accountEntry == null) {
+			throw new NoSuchEntryException(
+				"Unable to find account with external reference code " +
+					externalReferenceCode);
+		}
```

Comment:
for the account do the same check

Reply:
@smottal for the accountEntryService it's there in the local service but not in remote, so I think I can easily add it and then switch to getAEBERC from fetchAEBERC.